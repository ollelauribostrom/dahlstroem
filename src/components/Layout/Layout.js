import React from 'react';
import PropTypes from 'prop-types';
import { StaticQuery, graphql } from 'gatsby';

import Header from '../Header';

const Layout = ({ children, path, color, menuBackground, menuColor }) => (
  <StaticQuery
    query={graphql`
      query SiteTitleQuery {
        site {
          siteMetadata {
            title
          }
        }
      }
    `}
    render={data => (
      <div id="outer-container">
        <Header
          siteTitle={data.site.siteMetadata.title}
          path={path}
          color={color}
          menuBackground={menuBackground}
          menuColor={menuColor}
        />
        <div className="content-wrapper">{children}</div>
      </div>
    )}
  />
);

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};

Layout.defaultProps = {
  color: { hex: '#000' },
  menuColor: { hex: '#fff' },
  menuBackground: { hex: '#000' },
};

export default Layout;
