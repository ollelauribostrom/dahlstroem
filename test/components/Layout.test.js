import React from 'react';
import { shallow } from 'enzyme';

import Layout from '../../src/components/Layout';

describe('Tests for Layout', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Layout
        location={{ pathname: '/test' }}
        color={{ hex: '#fff' }}
        menuColor={{ hex: '#fff' }}
        menuBackground={{ hex: '#000' }}
      >
        <span>childA</span>
        <span>childB</span>
      </Layout>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('uses defaultProps values for colors if none is provided', () => {
    const wrapper = shallow(
      <Layout location={{ pathname: '/test' }}>
        <span>child</span>
      </Layout>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
