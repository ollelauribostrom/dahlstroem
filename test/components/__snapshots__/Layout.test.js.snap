// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Layout renders correctly 1`] = `
<mockConstructor
  render={[Function]}
/>
`;

exports[`Tests for Layout uses defaultProps values for colors if none is provided 1`] = `
<mockConstructor
  render={[Function]}
/>
`;
